StyleCop.Analyzers/StyleCop.Analyzers/NamingRules/SA1307AccessibleFieldsMustBeginWithUpperCaseLetter.cs

```diff
@@ -49,7 +54,31 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxNodeAction(this.HandleFieldDeclarator, SyntaxKind.FieldDeclaration);
+        }
+
+        private void HandleFieldDeclarator(SyntaxNodeAnalysisContext context)
+        {
+            // To improve performance we are looking for the field instead of the declarator directly. That way we don't get called for local variables.
+            FieldDeclarationSyntax declaration = context.Node as FieldDeclarationSyntax;
+            if (declaration != null && declaration.Declaration != null)
+            {
+                if (declaration.Modifiers.Any(SyntaxKind.PublicKeyword) || declaration.Modifiers.Any(SyntaxKind.InternalKeyword))
+                {
+                    foreach (VariableDeclaratorSyntax declarator in declaration.Declaration.Variables)
+                    {
+                        string name = declarator.Identifier.ToString();
+
+                        if (!string.IsNullOrEmpty(name) 
+                            && !char.IsUpper(name[0]) 
+                            && char.IsLetter(name[0])
```

Comment:
As discussed in #369, this check should use `char.IsLower(name[0])` instead of the current condition.